### PR TITLE
Correct doc for WorkQueue<T>::pop().

### PR DESCRIPTION
**Pull request by nieksand**
The old function doc looks like copy-pasta from WorkQueue::insert().

WorkQueue::pop() does not enqueue nor does it return a boolean false.  Doc corrected accordingly.

### CODE REVIEW

File: src/librustc_data_structures/work_queue.rs [hunk 1]
```diff
@@ -53,7 +53,7 @@ impl<T: Idx> WorkQueue<T> {
         }
     }
 
-    /// Attempt to enqueue `element` in the work queue. Returns false if it was already present.
+    /// Attempt to pop element from the work queue.
```
Comment (varkor) [inline src/librustc_data_structures/work_queue.rs]: Nit: "an element".

Comment (nieksand) [reply src/librustc_data_structures/work_queue.rs]: Repushed with change.

Change is in separate commit.  Let me know if I should squash it with rebase instead.

Comment (varkor) [reply src/librustc_data_structures/work_queue.rs]: Thanks — if you could squash instead, that would be great!

**Comment by nieksand** [comment]
blergh.  Just screwed that up.  Will untangle the git mess in a bit.

**Comment by nieksand** [comment]
Cleaned up the git mess.  Single commit with change.

**Comment by nieksand** [comment]
Thanks for reviewing

**Pull request merged**